testapp/build.gradle

```diff
@@ -45,18 +45,23 @@ android {
             appId          : property("HOST_APP_PROD_ID") ?: "test-host-app-id",
             subscriptionKey: property("HOST_APP_PROD_SUBSCRIPTION_KEY") ?: "test-subs-key"
     ]
+
+    def buildVersion = new Date().format('yyMMddHHmm')
```

Comment:
Maybe you could use the Circle CI build number (`CIRCLE_BUILD_NUM` env variable see https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables), and then use a timestamp only if the env variable isn't set? Something like `System.getenv("CIRCLE_BUILD_NUM") ?: new Date().format('yyMMddHHmm')`